File: TiaCodegen/Commands/Move.cs
using System.Linq;
using TiaCodegen.Interfaces;

namespace TiaCodegen.Commands
{
    public class Move : BaseOperationOrSignal, IFunctionOperation
    {
        public Move(params IOperationOrSignal[] operationOrSignals)
            : base(operationOrSignals)
        { }

        public override int Cardinality { get { return Children.Count() - 1; } }
    }
}

File: TiaCodegen-ts/src/Commands/Move.ts
import { IOperationOrSignal } from '../Interfaces/IOperationOrSignal.js';
import { IFunctionOperation } from '../Interfaces/IFunctionOperation.js';
import { BaseOperationOrSignal } from './BaseOperationOrSignal.js';

export class Move extends BaseOperationOrSignal implements IFunctionOperation {
    constructor(...operationOrSignals: IOperationOrSignal[]) {
        super(...operationOrSignals);
    }

    get cardinality(): number {
        return this.children.length - 1;
    }
    set cardinality(_value: number) {
        // readonly via getter, ignore sets
    }
}
